### PR TITLE
Take screenshot in dialogs #13184

### LINKED ISSUE #9508: Map Editor: Screenshot hotkey doesn't work
there's no way to make a screenshot of the Lobby either
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/27122662-map-editor-screenshot-hotkey-doesn-t-work?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #13184: Screenshot not working in lobby
Should also work there, and in all menues

### PR DESCRIPTION
**Pull request by rob-v**
Closes #13184
Closes #9508
Take screenshot works fine ingame or in main menu, doesn't work usually in lobby and all dialogs.
I was thinking if this is needed/worth, but the issue is that:
1. Take screenshot works e.g. in lobby or any tab/dialog until you activate Text field. Then it is not possible to take screenshot without moving to another tab/window and back (clicking another widget like checkbox on the same tab doesn't help). So it works at beginning but then stops to work so obviously looks like/is bug.
2. You can take screenshot from Windows even while in Textbox

This small change enables screenshot also in lobby/dialogs anytime.
Note: it is small code duplication. it could be done only once in Ui.HandleKeyPress with little perf impact.

### CODE REVIEW

**Review by pchote — COMMENTED** [review]
This seems like the most reasonable solution we can do, and IMO is an improvement over the hardcoded `RootWidget` class.

Can you please move over the rest of the `RootWidget` keys, and then replace that with a normal widget?

**Comment by rob-v** [comment]
Updated. 
I was thinking about replacing Down by Up or Down with MultiTapCount (#13194) for these rare/longer operations.

File: OpenRA.Game/Widgets/Widget.cs [hunk 1]
```diff
@@ -128,6 +128,31 @@ public static bool HandleInput(MouseInput mi)
...
+
+				if (hk == Game.Settings.Keys.DevReloadChromeKey)
+				{
+					ChromeProvider.Initialize(Game.ModData);
+					return true;
+				}
+
+				if (hk == Game.Settings.Keys.HideUserInterfaceKey)
+				{
+					foreach (var child in Root.Children)
```
Comment (pchote) [inline OpenRA.Game/Widgets/Widget.cs]: This can now simplify to `Root.Visible ^= true;` now that the root widget isn't handling the input.

**Review by pchote — APPROVED** [review]
LGTM.  :+1: after the following fix

**Comment by rob-v** [comment]
Updated.

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/b61502b0e1476de5ee105a372dd07de4c624b9d9...192ed715b7dac40eb39c655a6d83adf35d795cc8)

**Pull request merged**